src/api/cpp/nixl.h

```diff
@@ -337,6 +337,19 @@ class nixlAgent {
         void
         releaseGpuXferReq(nixlGpuXferReqH *gpu_req_hndl) const;
 
+        /**
+         * @brief  Prepare a signal for GPU transfer.
+         *
+         * The caller must allocate and register the signal memory before calling this function.
+         * Use @ref getBackendParams to query the required signal size, allocate
```

Comment:
This is a misuse of an existing API. 'getBackendParams` returns a list of initialization parameters and their values for the backend. This method can be used to get the default values used for parameters that were not specified by the user.
The size of the device counter is not controlled by the user.

It seems that NIXL has to prepare the memory for the signal itself. It might be worth preparing it along with the creation of `nixlGpuXferReqH`. Otherwise, we'll just have to repeat the UCX API (query signal size + initialize signal) in the `nixlBackendEngine`, and in `nixlAgent`.